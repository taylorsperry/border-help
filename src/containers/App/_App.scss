@import '../../variables';

.App {
  background-color: $white;
}

.header {
  display: flex;
  align-items: center;
  height: 100px;
  padding: 0px 20px;
}

.logo {
  color: $blue;
  font-family: 'Anton';
  margin: 0px;
  font-size: 2.5rem;
}

.container {
  display: flex;
  justify-content: center;
}

