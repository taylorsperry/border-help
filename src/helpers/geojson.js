export const coordinates = 
  [
    [
      -117.0703125,
      32.69486597787505
    ],
    [
      -115.57617187499999,
      32.69486597787505
    ],
    [
      -114.873046875,
      32.69486597787505
    ],
    [
      -113.73046875,
      32.24997445586331
    ],
    [
      -111.62109375,
      31.50362930577303
    ],
    [
      -110.12695312499999,
      31.50362930577303
    ],
    [
      -108.28125,
      31.27855085894653
    ],
    [
      -108.10546875,
      31.728167146023935
    ],
    [
      -106.5234375,
      31.728167146023935
    ],
    [
      -105.99609375,
      31.50362930577303
    ],
    [
      -104.94140625,
      30.524413269923986
    ],
    [
      -104.58984375,
      29.84064389983441
    ],
    [
      -103.447265625,
      29.152161283318915
    ],
    [
      -102.65625,
      29.305561325527698
    ],
    [
      -102.216796875,
      29.76437737516313
    ],
    [
      -101.25,
      29.76437737516313
    ],
    [
      -100.72265625,
      28.92163128242129
    ],
    [
      -99.84374999999999,
      27.839076094777816
    ],
    [
      -99.404296875,
      27.059125784374068
    ],
    [
      -98.525390625,
      26.27371402440643
    ],
    [
      -97.734375,
      26.115985925333536
    ],
    [
      -97.294921875,
      25.958044673317843
    ]
  ]
