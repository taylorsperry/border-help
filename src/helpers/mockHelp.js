export const mockHelp = [
  {
    id: "help1",
    title: "Freedom for Immigrants",
    detail: [
      {
        para: "Paragraph one"
      },
    ]
  }
]