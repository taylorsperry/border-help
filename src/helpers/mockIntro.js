export const mockIntro = [
  {
    para: "Paragraph one"
  },
  {
    para: "Paragraph two"
  }
]