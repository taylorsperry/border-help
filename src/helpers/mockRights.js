export const mockRights = [
  {
    id: 1, 
    title: "You have the right to remain silent",
    detail: [
      {
        para: "Paragraph one"
      },
      {
        para: "Paragraph two"
      }
    ]
  },
  {
    id: 2, 
    title: "You cannot be detained without reasonable suspicion",
    detail: [
      {
        para: "Paragraph one"
      },
      {
        para: "Paragraph two"
      }
    ]
  }
]