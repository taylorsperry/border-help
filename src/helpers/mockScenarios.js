export const mockScenarios = [
  {
    id: "a",
    title: "You are asked for documents",
    detail: [
      {
        para: "Paragraph one",
      },
      {
        para: "Paragraph two",
      }
    ]
  },
  {
    id: "b",
    title: "You are stopped on a bus or a train",
    detail: [
      {
        para: "Paragraph one",
      },
      {
        para: "Paragraph two",
      }
    ]
  }
]