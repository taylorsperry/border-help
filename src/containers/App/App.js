import React, { Component } from 'react';
import { withRouter, Route, Link, Switch } from 'react-router-dom'
import { connect } from 'react-redux'
import './_App.scss';
import Intro from '../../components/Intro/Intro'
import Nav from '../../components/Nav/Nav'
import Category from '../../components/Category/Category'
import Location from '../Location/Location'
import { fetchRights } from '../../thunks/fetchRights'
import { fetchScenarios } from '../../thunks/fetchScenarios'
import { fetchHelp } from '../../thunks/fetchHelp'
import { fetchIntro } from '../../thunks/fetchIntro'
import Error from '../../components/Error/Error'
import PropTypes from 'prop-types'

export class App extends Component {

 async componentDidMount() {
    const url = 'http://localhost:3001/api/v1/intro'
    await this.props.fetchIntro(url)
  }

  getRights = () => {
    const url = 'http://localhost:3001/api/v1/rights/'
    const { history, rights } = this.props
    if (!rights.length) {
      this.props.fetchRights(url)
    } else {
      history.push('/rights')
    }
  }
  
  getScenarios = () => {
    const url = 'http://localhost:3001/api/v1/scenarios'
    const { history, scenarios } = this.props
    if (!scenarios.length) {
      this.props.fetchScenarios(url)
    } else {
      history.push('/scenarios')
    }
  }

  getHelp = () => {
    const url = 'http://localhost:3001/api/v1/help'
    const { history, help } = this.props
    if (!help.length) {
      this.props.fetchHelp(url)
    } else {
      history.push('/help')
    }
  }

  render() {
    const { intro, rights, scenarios, help } = this.props
    return (
      <div className="App">
        <header className='header'>
          <Link to='/'
                className='header-link'
          >
            <h1 className='logo'>BorderHelp</h1>
          </Link>
        </header>
          <Route 
            path='/'
            render={() => <Nav getRights={this.getRights} getScenarios={this.getScenarios} getHelp={this.getHelp} />}
          />
          <div className='container'>
          <Switch>
            <Route 
              exact path='/'
              render={() => <Intro data={intro}/>}
              />
            <Route 
              exact path='/rights'
              render={() => <Category key='rights' catName='Your Rights' data={rights} callFetch={this.getRights} />}
            />
            <Route 
              exact path='/what-to-do'
              render={() => <Category key='scen' catName='What to do if . . . ' data={scenarios} callFetch={this.getScenarios} />}
            />
            <Route 
              exact path='/help'
              render={() => <Category key='help' catName='Help' data={help} callFetch={this.getHelp} />}
            />
            <Route 
              exact path='/location'
              render={() => <Location />} 
            />
            <Route component={Error} />
          </Switch>
        </div>
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  intro: state.intro,
  rights: state.rights,
  scenarios: state.scenarios,
  help: state.help,
})

export const mapDispatchToProps = (dispatch) => ({
  fetchIntro: (url) => dispatch(fetchIntro(url)),
  fetchRights: (url) => dispatch(fetchRights(url)),
  fetchScenarios: (url) => dispatch(fetchScenarios(url)),
  fetchHelp: (url) => dispatch(fetchHelp(url))
})

App.propTypes = {
  intro: PropTypes.array,
  rights: PropTypes.array,
  scenarios: PropTypes.array,
  help: PropTypes.array,
  fetchIntro: PropTypes.func,
  fetchRights: PropTypes.func,
  fetchScenarios: PropTypes.func,
  fetchHelp: PropTypes.func
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App));
