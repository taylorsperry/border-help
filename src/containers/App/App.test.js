import React from 'react';
import { App } from './App';
import { shallow } from 'enzyme';
// import { storeRights, storeScenarios } from '../../actions'
import { mockIntro } from '../../helpers/mockIntro'
import { mockRights } from '../../helpers/mockRights'
import { mockScenarios } from '../../helpers/mockScenarios'
import { mockHelp } from '../../helpers/mockHelp'

jest.mock('../../actions')


describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('componentDidMount', () => {

    it('should call fetch and return an array of introductory paragraphs', async () => {
      const mockProps = {
        storeIntro: jest.fn()
      }

      wrapper = shallow(<App {...mockProps} />)

      window.fetch = jest.fn().mockImplementation( () => Promise.resolve({
        ok: true,
        status: 200,
        json: () => Promise.resolve(mockIntro)
      }))

      wrapper.instance().componentDidMount()
      const response = await window.fetch()
      const intro = await response.json()

      expect(intro).toEqual(mockIntro)
    })
  })

  describe('getRights', () => {
    
    it('should call fetchRights if there are no rights in the store', () => {
      const emptyRights = []
      wrapper = shallow(<App rights={emptyRights} />)
      const spy = jest.spyOn(wrapper.instance(), 'fetchRights')

      wrapper.instance().getRights()
      
      expect(spy).toHaveBeenCalled()
    })

    it('should not call fetchRights if there are rights in the store', () => {
      const mockProps = {
        rights: mockRights,
        history: { push: jest.fn() }
      }
      wrapper = shallow(<App {...mockProps} />)
      const spy = jest.spyOn(wrapper.instance(), 'fetchRights')

      wrapper.instance().getRights()

      expect(spy).not.toHaveBeenCalled()
    })
  })

  describe('getScenarios', () => {
    
    it('should call fetchScenarios if there are no scenarios in the store', () => {
      const emptyScenarios = []
      const mockProps = {
        scenarios: emptyScenarios,
        history: { push: jest.fn() }

      }
      wrapper = shallow(<App {...mockProps} />)
      const spy = jest.spyOn(wrapper.instance(), 'fetchScenarios')

      wrapper.instance().getScenarios()

      expect(spy).toHaveBeenCalled()
    })

    it('should not call fetchScenarios if there are scenarios in the store', () => {
      const mockProps = {
        scenarios: mockScenarios,
        history: { push: jest.fn() }
      }
      wrapper = shallow(<App {...mockProps} />)
      const spy = jest.spyOn(wrapper.instance(), 'fetchScenarios')

      wrapper.instance().getScenarios()

      expect(spy).not.toHaveBeenCalled()
    })
  })

  describe('getHelp', () => {
    
    it('should call fetchHelp if there is no help in the store', () => {
      const emptyHelp = []
      const mockProps = {
        help: emptyHelp,
        history: { push: jest.fn() }
      }
      wrapper = shallow(<App {...mockProps} />)
      const spy = jest.spyOn(wrapper.instance(), 'fetchHelp')

      wrapper.instance().getHelp()

      expect(spy).toHaveBeenCalled()
    })

    it('should not call fetchHelp if there is help in the store', () => {
      const mockProps = {
        help: mockHelp,
        history: { push: jest.fn() }
      }
      wrapper = shallow(<App {...mockProps} />)
      const spy = jest.spyOn(wrapper.instance(), 'fetchHelp')

      wrapper.instance().getHelp()

      expect(spy).not.toHaveBeenCalled()
    })
  })

  describe('fetchRights', async () => {

    beforeEach(() => {
      const mockProps = {
        storeRights: jest.fn()
      }
      
      wrapper = shallow(<App {...mockProps} />)
      
      window.fetch = jest.fn().mockImplementation( () => Promise.resolve({
        ok: true,
        status: 200,
        json: () => Promise.resolve(mockRights)
      }))
    })

    it('should call fetch and return an array of rights', async () => {

      wrapper.instance().fetchRights()

      const response = await window.fetch()
      const rights = await response.json()
      
      expect(window.fetch).toHaveBeenCalled()
      expect(rights).toEqual(mockRights)
    })

    it.skip('should dispatch storeRights with the response from the fetch', async () => {
      const response = await window.fetch()
      const rights = await response.json()
      
      expect(window.fetch).toHaveBeenCalled()
      expect(wrapper.props.storeRights).toHaveBeenCalledWith(rights)
    })
  })

  describe('fetchScenarios', () => {

    beforeEach(() => {
      const mockProps = {
        storeScenarios: jest.fn()
      }

      wrapper = shallow(<App {...mockProps} />)

      window.fetch = jest.fn().mockImplementation( () => Promise.resolve({
        ok: true,
        status: 200,
        json: () => Promise.resolve(mockScenarios)
      }))
    })

    it('should call fetch and return an array of scenarios', async () => {
      
      wrapper.instance().fetchScenarios()
      const response = await window.fetch()
      const scenarios = await response.json()

      expect(scenarios).toEqual(mockScenarios)
    })

    it.skip('should dispatch storeScenarios with the response from the fetch', () => {

    })
  })

  describe('fetchHelp', () => {

    beforeEach(() => {
      const mockProps = {
        storeHelp: jest.fn()
      }

      wrapper = shallow(<App {...mockProps} />)

      window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
        ok: true,
        status: 200,
        json: () => Promise.resolve(mockHelp)
      }))
    })

    it('should call fetch and return an array of help', async () => {

      wrapper.instance().fetchHelp()
      const response = await window.fetch()
      const help = await response.json()

      expect(help).toEqual(mockHelp)
    })
  })

})

