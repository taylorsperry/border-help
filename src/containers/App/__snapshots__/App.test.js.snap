// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App getRights should call fetchRights if there are no rights in the store 1`] = `
<Route>
  <Component />
</Route>
`;

exports[`App should match the snapshot 1`] = `
<Route>
  <Component />
</Route>
`;
